presto-main/src/main/java/com/facebook/presto/sql/planner/SubfieldUtils.java

```diff
@@ -53,6 +54,16 @@ public static SubfieldPath deferenceOrSubscriptExpressionToPath(Node expression)
             else if (expression instanceof SubscriptExpression) {
                 SubscriptExpression subscript = (SubscriptExpression) expression;
                 Expression index = subscript.getIndex();
+                if (index instanceof Cast) {
+                    // map['str'] can be represented as a
+                    // SubscriptExpression with an index of cast of a
+                    // string literal to a varchar(xx).
+                    Cast cast = (Cast) index;
+                    index = cast.getExpression();
+                    if (!(index instanceof StringLiteral) || !cast.getType().startsWith("varchar")) {
+                        return null;
+                    }
```

Comment:
- Can this block be simplified to `index = ((Cast) index).getExpression();`?
- Would you update TestLogicalPlanner#testPushdownSubfields to add a query that hits that path?

```
        assertPlanWithSession(
                "SELECT string_to_int_map['foo'] FROM lineitem_ext",
                pushdownSubfields, false,
                anyTree(
                        project(
                                ImmutableMap.of("map_filtered", expression(format("filter_by_subscript_paths(string_to_int_map, %s)[cast('foo' as varchar)]", optimizeExpression("array['string_to_int_map[\"foo\"]']", new ArrayType(createVarcharType(24)))))),
                                tableScan("lineitem_ext", ImmutableMap.of("string_to_int_map", "string_to_int_map")))));
```

This would require adding a new column to lineitem_ext in TpchMetadata:

```
                                .add(new ColumnMetadata("string_to_int_map", new MapType(
                                        VARCHAR,
                                        BIGINT,
                                        methodHandle(TpchMetadata.class, "throwUnsupportedOperation"),
                                        methodHandle(TpchMetadata.class, "throwUnsupportedOperation"),
                                        methodHandle(TpchMetadata.class, "throwUnsupportedOperation"),
                                        methodHandle(TpchMetadata.class, "throwUnsupportedOperation"))))
...
    public static void throwUnsupportedOperation()
    {
        throw new UnsupportedOperationException();
    }
```